### DISCUSSION TITLE
Cannot draw colors in a viewport

### DISCUSSION THREAD

**Discussion by rohit-kumar-j** [google-deepmind/mujoco#1966] [Asking for Help] [answered]
Hi,
I'm trying to modify the pixel values in a viewport to black out the image before mjr_drawPixels() is called but it does not seem to work:
```cpp
...
  mjv_updateScene(sim->m_, sim->d_, opt, NULL, &offscreen_cam, mjCAT_ALL, scn);
  mjr_render(viewport, &sim->scn, &sim->platform_ui->mjr_context());
  mjr_readPixels(color_buffer, nullptr, viewport, &sim->platform_ui->mjr_context());

  for(int i=0; i<VIEWPORT_HEIGHT; i++){
    for(int j=0; j<VIEWPORT_WIDTH; j++){
      int index = (i * VIEWPORT_WIDTH + j) * 3;
      color_buffer[index] = 0;      // Set Red component
      color_buffer[index + 1] = 0;  // Set Green component
      color_buffer[index + 2] = 0;  // Set Blue component
      // printf("[%u %u %u]", r,g,b);
    }
    // printf("\n");
  }
  // printf("\n");
  mjr_drawPixels(color_buffer, nullptr, viewport, &sim->platform_ui->mjr_context());
...
```

Instead it just only shows a viewport with the camera view in the corner:
### [Here](https://github.com/rohit-kumar-j/AUVC/blob/a21f47ebbfcc90478fb38cb13caf742db219d6c5/simulate/simulate.cc#L643) is the source for the modified `simulate.cc`
![image](https://github.com/user-attachments/assets/4abecb4f-0a83-49ff-8986-e04be97cf4f8)

**Comment by tenfoldpaper** [comment, marked as answer]
Hey Rohit, I've sort of figured out the issue [here](https://github.com/google-deepmind/mujoco/issues/744#issuecomment-2461979081); while the overlay is only broadcasting the depth version of the main camera, I think the logic will be helpful.
